gcp_variant_transforms/libs/processed_variant.py

```diff
@@ -235,51 +244,18 @@ def _add_annotation(self, proc_var, field_name, data):
     Args:
       proc_var (ProcessedVariant): The object to which the annotations are being
         added.
-      field_name (str): The name of the annotation field.
-      data(List[str]): The data part of the field separated on comma.
+      field_name: The name of the annotation field.
+      data: The data part of the field separated on comma.
     """
     if field_name not in self._header_fields.infos:
       raise ValueError('{} INFO not found in the header, variant: {}'.format(
           field_name, proc_var))
     header_desc = self._header_fields.infos[field_name].desc
     annotation_names = _extract_annotation_names(header_desc)
-    alt_annotation_map = self._convert_annotation_strs_to_alt_map(
-        annotation_names, data)
-    for alt_bases, annotations_list in alt_annotation_map.iteritems():
-      # This assumes that number of alternate bases and annotation segments
-      # are not too big. If this assumption is not true, we should replace the
-      # following loop with a hash table search and avoid the quadratic time.
-      for alt in proc_var._alternate_datas:
-        if alt.alternate_bases == alt_bases:
-          alt._info[field_name] = annotations_list
-          self._annotation_counter.inc()
-          break
-      # TODO(bashir2): Currently we only check exact matches of alternate bases
-      # which is not enough. We should implement the whole standard for finding
-      # alternate bases for an annotation list.
-      else:
-        self._annotation_alt_mismatch_counter.inc()
-        logging.warning('Could not find matching alternate bases for %s in '
-                        'annotation filed %s', alt_bases, field_name)
-
-  def _convert_annotation_strs_to_alt_map(self, annotation_names, field_data):
-    # type: (List[str], List[str]) -> Dict[str, List[Dict[str, str]]]
-    alt_annotation_map = defaultdict(list)
-    for annotation_str in field_data:
-      annotations = _extract_annotation_list_with_alt(annotation_str)
-      alt_annotation_map[annotations[0]].append(
-          self._create_annotation_map(annotations, annotation_names))
-    return alt_annotation_map
-
-  def _create_annotation_map(self, annotations, annotation_names):
-    # type: (List[str], List[str]) -> Dict[str, str]
-    if len(annotation_names) != len(annotations) - 1:
-      raise ValueError('Expected {} annotations, got {}'.format(
-          len(annotation_names), len(annotations) - 1))
-    annotation_dict = {}
-    for index, name in enumerate(annotation_names):
-      annotation_dict[name] = annotations[index + 1]
-    return annotation_dict
+    annotation_processor = _AnnotationProcessor(
```

Comment:
From the API of this class, it's not really clear why some args are in the constructor vs. in the add_annotation_data method (e.g. field_name and annotation_names). It's also not clear why a new instance of the AnnotationProcessor class is needed for every annotation. I think a better design may be to have a single AnnotationProcessor (that has counters) and you'd use that instance for processing any annotation? This paradigm also resolves the issue of why some fields would be in the constructor vs. in the methods.